src/psyclone/dynamo0p3.py

```diff
@@ -7141,7 +7141,10 @@ def lower_to_language_level(self):
             pos = self.position
             parent = self.parent
             self.detach()
-            for child in self.loop_body.pop_all_children():
+            all_children_reverse = self.loop_body.pop_all_children()[::-1]
```

Comment:
I've never seen this before. More pythonic to use "...children().reverse()" perhaps?

Reply:
You would think so, wouldn't you? I did, too 😁  But that doesn't work: the result of `reverse` is `None`, so `all_children_reverse` becomes `None`, not the reverse of the list. I actually tried that first - the only way I could get it work was like this:
```
all_children_reverse = self.loop_body.pop_all_children()
all_children_reverse.reverse()
```
And I just hated the look of that: we assign the non-reverse list to a variable called `all_children_reverse`.

I did some googling, and found `reversed`, so I've changed it to:
```
all_children_reverse = reversed(self.loop_body.pop_all_children())
```
I hope that's an acceptable compromise.

Reply:
Yep, that's great.